client/src/modules/reports/generate/open_debtors/open_debtors.config.js

```diff
@@ -0,0 +1,82 @@
+angular.module('bhima.controllers')
+  .controller('open_debtorsController', OpenDebtorsConfigController);
+
+OpenDebtorsConfigController.$inject = [
+  '$sce', 'NotifyService', 'BaseReportService', 'AppCache', 'reportData', '$state', '$translate',
+];
+
+function OpenDebtorsConfigController($sce, Notify, SavedReports, AppCache, reportData, $state, $translate) {
+  var vm = this;
+  var cache = new AppCache('configure_open_debtors');
+  var reportUrl = 'reports/finance/debtors/open';
+
+  vm.previewGenerated = false;
+  vm.reportDetails = {};
+
+  checkCachedConfiguration();
+
+  vm.clearPreview = function clearPreview() {
+    vm.previewGenerated = false;
+    vm.previewResult = null;
+  };
+
+  vm.columnOrder = [
+    {
+      label : $translate.instant('REPORT.ORDER.LAST_PAYMENT'),
+      value : 'payment-date',
+    },
+    {
+      label : $translate.instant('REPORT.ORDER.LAST_INVOICE'),
+      value : 'invoice-date',
+    },
+    {
+      label : $translate.instant('REPORT.ORDER.PATIENT_NAME'),
+      value : 'patient-name',
+    },
+    {
+      label : $translate.instant('REPORT.ORDER.TOTAL_DEBT'),
+      value : 'debt',
+    },    
+  ];
+
+  vm.requestSaveAs = function requestSaveAs() {
+    var options = {
+      url : reportUrl,
+      report : reportData,
+      reportOptions : angular.copy(vm.reportDetails),
+    };
+
+    return SavedReports.saveAsModal(options)
+      .then(function () {
+        $state.go('reportsBase.reportsArchive', { key : options.report.report_key });
+      })
+      .catch(Notify.handleError);
+  };
+
+  vm.preview = function preview(form) {
+    if (form.$invalid) { return; }
+    
+    var orderBy = vm.reportDetails.orderBy? '-asc' : '-desc';
+    vm.reportDetails.order += orderBy; 
```

Comment:
I propose that you do not modify `vm.reportDetails.order`.  It will destroy the `uiSelect`'s selection since the model will have changed.

Instead, create another object like this:
```js
var params = angular.copy(vm.reportDetails);
cache.reportDetails = angular.copy(vm.reportDetails);

var direction = vm.reportDetails.orderBy ? '-asc' : '-desc';
params.order += direction;

return SavedReports.requestPreview(reportUrl, reportData.id, params)
```

That way, when you click "Cancel Preview" the original data is preserved.